core/templates/dev/head/pages/exploration_editor/RouterServices.js

```diff
@@ -44,14 +44,20 @@ oppia.factory('routerService', [
       return $location.path();
     }, function(newPath, oldPath) {
       if (newPath === '') {
-        $location.path(oldPath);
+        if (oldPath === '/') {
+          $window.history.go(-1);
+        } else {
+          $location.path(oldPath);
+        }
         return;
       }
-
       if (!oldPath) {
         // This can happen when clicking on links whose href is "#".
         return;
       }
+      if (newPath === '/' && oldPath.includes('/gui/')) {
```

Comment:
What I meant was if you press the create exploration button on the library page then after creation of the exploration if you press the back button you must go back to the library page again. Hence the redirection problem is not specifically in the dashboard page as the url redirection flow when a card is clicked from dashboard or whenever a new exploration is created is /create/exploration_id then /create/exploration_id#/ then /create/exploration_id#/gui/(statename). 

If you copy/paste the link manually of the exploration you own then the back button redirection works fine.